src/deluge/model/mod_controllable/filters/filter_config.h

```diff
@@ -0,0 +1,145 @@
+constexpr FilterMode kLastLadder = FilterMode::TRANSISTOR_24DB_DRIVE;
+// Off is not an LPF mode but is used to reset filters
+constexpr int32_t kNumLPFModes = util::to_underlying(FilterMode::SVF_NOTCH) + 1;
+constexpr int32_t kFirstHPFMode = util::to_underlying(FilterMode::SVF_BAND);
+constexpr int32_t kNumHPFModes = util::to_underlying(FilterMode::OFF) - kFirstHPFMode;
+enum class FilterRoute {
+	HIGH_TO_LOW,
+	LOW_TO_HIGH,
+	PARALLEL,
+};
+
+constexpr int32_t kNumFilterRoutes = util::to_underlying(FilterRoute::PARALLEL) + 1;
+
+FilterRoute stringToFilterRoute(char const* string);
+char const* filterRouteToString(FilterRoute route);
+
+char const* lpfTypeToString(FilterMode lpfType);
+FilterMode stringToLPFType(char const* string);
+
+namespace deluge::dsp::filter {
+
+enum class FilterFamily { LP_LADDER, SVF, HP_LADDER, NONE };
+constexpr uint8_t kNumFamilies = 3;
+enum LpLadderType { LP12, LP24, DRIVE };
+constexpr uint8_t kNumLadders = 3;
+enum SVFType { BAND, NOTCH };
+constexpr uint8_t kNumSVF = 2;
+enum HPFType { HP12 };
+constexpr uint8_t kNumHPLadders = 1;
+
+using specificFilterType = uint8_t;
+// must match order of filter family declaration for indexing to work
+constexpr uint8_t numForVariant[kNumFamilies] = {kNumLadders, kNumSVF, kNumHPLadders};
+
+/// A filters family (types which can all share an implementation for toggling) and specific type within that family
+class SpecificFilter {
+public:
+	SpecificFilter() = default;
+	/// Construct a filtertype from a mode
+	SpecificFilter(FilterMode mode) {
+		switch (mode) {
+		case FilterMode::OFF:
+			family = FilterFamily::NONE;
+			type = LpLadderType::LP12;
+			break;
+		case FilterMode::HPLADDER:
+			family = FilterFamily::HP_LADDER;
+			type = HPFType::HP12;
+			break;
+		case FilterMode::SVF_BAND:
+			family = FilterFamily::SVF;
+			type = SVFType::BAND;
+			break;
+		case FilterMode::SVF_NOTCH:
+			family = FilterFamily::SVF;
+			type = SVFType::NOTCH;
+			break;
+		case FilterMode::TRANSISTOR_12DB:
+			family = FilterFamily::LP_LADDER;
+			type = LpLadderType::LP12;
+			break;
+		case FilterMode::TRANSISTOR_24DB:
+			family = FilterFamily::LP_LADDER;
+			type = LpLadderType::LP24;
+			break;
+		case FilterMode::TRANSISTOR_24DB_DRIVE:
+			family = FilterFamily::LP_LADDER;
+			type = LpLadderType::DRIVE;
+			break;
+		}
+	}
+
+	FilterFamily getFamily() { return family; }
+
+	specificFilterType getType() { return type; }
+
+	[[nodiscard]] l10n::String getMorphName() const {
+
+		switch (family) {
+		case FilterFamily::LP_LADDER:
+			return l10n::String::STRING_FOR_DRIVE;
+		case FilterFamily::HP_LADDER:
+			return l10n::String::STRING_FOR_FM;
+		case FilterFamily::SVF:
+			return l10n::String::STRING_FOR_MORPH;
+		default:
+			return l10n::String::STRING_FOR_NONE;
+		};
+	};
+
+	SpecificFilter& incrementMode() {
+		type = (type + 1) % numForVariant[util::to_underlying(family)];
+		return *this;
+	} // namespace deluge::dsp::filter
```

Comment:
Old comment?

Reply:
No, it's in the filter namespace intentionally. It's only in the models folder so that it compiles as thumb. It's not much code so probably wouldn't make a big difference to move back though